Code/GraphMol/RingInfo.h

```diff
@@ -13,6 +13,14 @@
 
 #include <map>
 #include <vector>
+#include <RDGeneral/BoostStartInclude.h>
+#include <boost/shared_ptr.hpp>
+#include <RDGeneral/BoostEndInclude.h>
+#ifdef RDK_USE_URF
+#include <RingDecomposerLib.h>
+#else
+typedef void RDL_data;
```

Comment:
I can't tell if this is clever or just evil.

This boils down to:

```
  std::shared_ptr<void> foo;
  std::shared_ptr<void> bar(foo);
```

Which is really, really weird.  You can't actually construct a foo that is non null as void isn't a data type except for bookkeeping.   I.e. you can't do

```
 void f;
```

It works here because of type erasure.

As it turns out, any empty struct or void * would work here and might make me cringe just a little bit less.

However, the other side says just put the dp_urfData into the conditional compilation with the rest of the code and save the 40 bytes of overhead from the shared_ptr :)